### PR TITLE
HBASE-27208 Use spotless to purge the missing summary warnings from error prone

### PR DESCRIPTION
**Pull request by Apache9**

### CODE REVIEW

**Comment by Apache9** [comment]
The first commit is what I added into the pom ,the second commit is the changes after running spotless:apply.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 49s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  2s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 18s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   2m 18s |  master passed  |
| +1 :green_heart: |  compile  |   6m  0s |  master passed  |
| +1 :green_heart: |  checkstyle  |   1m  1s |  master passed  |
| +1 :green_heart: |  spotless  |   0m 40s |  branch has no errors when running spotless:check.  |
| +1 :green_heart: |  spotbugs  |  13m 24s |  master passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 14s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   2m  6s |  the patch passed  |
| +1 :green_heart: |  compile  |   5m 58s |  the patch passed  |
| -0 :warning: |  javac  |   5m 58s |  root generated 182 new + 1108 unchanged - 268 fixed = 1290 total (was 1376)  |
| +1 :green_heart: |  checkstyle  |   1m  1s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  xml  |   0m  1s |  The patch has no ill-formed XML file.  |
| +1 :green_heart: |  hadoopcheck  |  11m 50s |  Patch does not cause any errors with Hadoop 3.1.2 3.2.2 3.3.1.  |
| +1 :green_heart: |  spotless  |   0m 36s |  patch has no errors when running spotless:check.  |
| +1 :green_heart: |  spotbugs  |  14m 55s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   1m 48s |  The patch does not generate ASF License warnings.  |
|  |   |  71m 23s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4628/1/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/4628 |
| Optional Tests | dupname asflicense javac spotbugs hadoopcheck hbaseanti spotless checkstyle compile xml |
| uname | Linux 1a01b84f6728 5.4.0-1071-aws #76~18.04.1-Ubuntu SMP Mon Mar 28 17:49:57 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 02f26368e2 |
| Default Java | AdoptOpenJDK-1.8.0_282-b08 |
| javac | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4628/1/artifact/yetus-general-check/output/diff-compile-javac-root.txt |
| Max. process+thread count | 139 (vs. ulimit of 30000) |
| modules | C: hbase-common hbase-hadoop-compat hbase-client hbase-zookeeper hbase-balancer hbase-http hbase-asyncfs hbase-procedure hbase-server hbase-mapreduce hbase-testing-util hbase-thrift hbase-shell hbase-it hbase-rest . U: . |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4628/1/console |
| versions | git=2.17.1 maven=3.6.3 spotbugs=4.2.2 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 46s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  3s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 18s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   2m 40s |  master passed  |
| +1 :green_heart: |  compile  |   1m 43s |  master passed  |
| +1 :green_heart: |  shadedjars  |   3m 53s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   4m 52s |  master passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 14s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   2m 30s |  the patch passed  |
| +1 :green_heart: |  compile  |   1m 44s |  the patch passed  |
| +1 :green_heart: |  javac  |   1m 44s |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   3m 52s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   4m 56s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  | 255m 56s |  root in the patch passed.  |
|  |   | 288m  7s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4628/1/artifact/yetus-jdk11-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/4628 |
| Optional Tests | javac javadoc unit shadedjars compile |
| uname | Linux 452e457aed23 5.4.0-1071-aws #76~18.04.1-Ubuntu SMP Mon Mar 28 17:49:57 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 02f26368e2 |
| Default Java | AdoptOpenJDK-11.0.10+9 |
|  Test Results | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4628/1/testReport/ |
| Max. process+thread count | 4784 (vs. ulimit of 30000) |
| modules | C: hbase-common hbase-hadoop-compat hbase-client hbase-zookeeper hbase-balancer hbase-http hbase-asyncfs hbase-procedure hbase-server hbase-mapreduce hbase-testing-util hbase-thrift hbase-shell hbase-it hbase-rest . U: . |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4628/1/console |
| versions | git=2.17.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 45s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  4s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 19s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   2m 17s |  master passed  |
| +1 :green_heart: |  compile  |   1m 27s |  master passed  |
| +1 :green_heart: |  shadedjars  |   3m 54s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   3m 54s |  master passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 13s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   2m  6s |  the patch passed  |
| +1 :green_heart: |  compile  |   1m 27s |  the patch passed  |
| +1 :green_heart: |  javac  |   1m 27s |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   3m 56s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   3m 56s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  | 386m 13s |  root in the patch passed.  |
|  |   | 415m 24s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4628/1/artifact/yetus-jdk8-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/4628 |
| Optional Tests | javac javadoc unit shadedjars compile |
| uname | Linux 7a4ae317f239 5.4.0-1071-aws #76~18.04.1-Ubuntu SMP Mon Mar 28 17:49:57 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 02f26368e2 |
| Default Java | AdoptOpenJDK-1.8.0_282-b08 |
|  Test Results | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4628/1/testReport/ |
| Max. process+thread count | 5064 (vs. ulimit of 30000) |
| modules | C: hbase-common hbase-hadoop-compat hbase-client hbase-zookeeper hbase-balancer hbase-http hbase-asyncfs hbase-procedure hbase-server hbase-mapreduce hbase-testing-util hbase-thrift hbase-shell hbase-it hbase-rest . U: . |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4628/1/console |
| versions | git=2.17.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 41s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  2s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 39s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   2m  5s |  master passed  |
| +1 :green_heart: |  compile  |   5m 57s |  master passed  |
| +1 :green_heart: |  checkstyle  |   1m  3s |  master passed  |
| +1 :green_heart: |  spotless  |   0m 39s |  branch has no errors when running spotless:check.  |
| +1 :green_heart: |  spotbugs  |  13m 21s |  master passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 12s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   2m  8s |  the patch passed  |
| +1 :green_heart: |  compile  |   5m 54s |  the patch passed  |
| -0 :warning: |  javac  |   5m 54s |  root generated 182 new + 1108 unchanged - 268 fixed = 1290 total (was 1376)  |
| -0 :warning: |  checkstyle  |   1m  3s |  root: The patch generated 1 new + 547 unchanged - 1 fixed = 548 total (was 548)  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  xml  |   0m  0s |  The patch has no ill-formed XML file.  |
| +1 :green_heart: |  hadoopcheck  |  11m 26s |  Patch does not cause any errors with Hadoop 3.1.2 3.2.2 3.3.1.  |
| +1 :green_heart: |  spotless  |   0m 35s |  patch has no errors when running spotless:check.  |
| +1 :green_heart: |  spotbugs  |  15m  0s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   1m 46s |  The patch does not generate ASF License warnings.  |
|  |   |  70m 51s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4628/2/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/4628 |
| Optional Tests | dupname asflicense javac spotbugs hadoopcheck hbaseanti spotless checkstyle compile xml |
| uname | Linux 4837f1bfde58 5.4.0-1071-aws #76~18.04.1-Ubuntu SMP Mon Mar 28 17:49:57 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 02f26368e2 |
| Default Java | AdoptOpenJDK-1.8.0_282-b08 |
| javac | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4628/2/artifact/yetus-general-check/output/diff-compile-javac-root.txt |
| checkstyle | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4628/2/artifact/yetus-general-check/output/diff-checkstyle-root.txt |
| Max. process+thread count | 138 (vs. ulimit of 30000) |
| modules | C: hbase-common hbase-hadoop-compat hbase-client hbase-zookeeper hbase-balancer hbase-http hbase-asyncfs hbase-procedure hbase-server hbase-mapreduce hbase-testing-util hbase-thrift hbase-shell hbase-it hbase-rest . U: . |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4628/2/console |
| versions | git=2.17.1 maven=3.6.3 spotbugs=4.2.2 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m  1s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  5s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 40s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   4m 35s |  master passed  |
| +1 :green_heart: |  compile  |   3m  8s |  master passed  |
| +1 :green_heart: |  shadedjars  |   6m  8s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |  11m 37s |  master passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 13s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   2m 31s |  the patch passed  |
| +1 :green_heart: |  compile  |   1m 43s |  the patch passed  |
| +1 :green_heart: |  javac  |   1m 43s |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   3m 51s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   5m 53s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  | 271m 41s |  root in the patch passed.  |
|  |   | 322m 16s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4628/2/artifact/yetus-jdk11-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/4628 |
| Optional Tests | javac javadoc unit shadedjars compile |
| uname | Linux f328884abcbc 5.4.0-1071-aws #76~18.04.1-Ubuntu SMP Mon Mar 28 17:49:57 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 02f26368e2 |
| Default Java | AdoptOpenJDK-11.0.10+9 |
|  Test Results | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4628/2/testReport/ |
| Max. process+thread count | 4701 (vs. ulimit of 30000) |
| modules | C: hbase-common hbase-hadoop-compat hbase-client hbase-zookeeper hbase-balancer hbase-http hbase-asyncfs hbase-procedure hbase-server hbase-mapreduce hbase-testing-util hbase-thrift hbase-shell hbase-it hbase-rest . U: . |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4628/2/console |
| versions | git=2.17.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 47s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  3s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 31s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   2m  9s |  master passed  |
| +1 :green_heart: |  compile  |   1m 25s |  master passed  |
| +1 :green_heart: |  shadedjars  |   3m 58s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   3m 57s |  master passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 12s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   2m  5s |  the patch passed  |
| +1 :green_heart: |  compile  |   1m 25s |  the patch passed  |
| +1 :green_heart: |  javac  |   1m 25s |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   3m 52s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   3m 56s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  | 387m 35s |  root in the patch passed.  |
|  |   | 416m 40s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4628/2/artifact/yetus-jdk8-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/4628 |
| Optional Tests | javac javadoc unit shadedjars compile |
| uname | Linux c35cc6bf0892 5.4.0-1071-aws #76~18.04.1-Ubuntu SMP Mon Mar 28 17:49:57 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 02f26368e2 |
| Default Java | AdoptOpenJDK-1.8.0_282-b08 |
|  Test Results | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4628/2/testReport/ |
| Max. process+thread count | 4904 (vs. ulimit of 30000) |
| modules | C: hbase-common hbase-hadoop-compat hbase-client hbase-zookeeper hbase-balancer hbase-http hbase-asyncfs hbase-procedure hbase-server hbase-mapreduce hbase-testing-util hbase-thrift hbase-shell hbase-it hbase-rest . U: . |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4628/2/console |
| versions | git=2.17.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache9** [comment]
@apurtell @virajjasani Please take a look at this one? It is used fix some MissingSummary error prone warnings, by a spotless rule, so we do not need to do it manually.

Thanks.

File: pom.xml [hunk 1]
```diff
@@ -2715,6 +2715,27 @@
...
+              <name>Purge single returns tag multi line</name>
+              <searchRegex>(?m)^ */\*\*\n *\* *@return *(.*) *\n *\*/$</searchRegex>
+              <replacement>/** Returns $1 */</replacement>
+            </replaceRegex>
+            <replaceRegex>
+              <name>Purge single returns tag single line</name>
+              <searchRegex>^ */\*\* *@return *(.*) *\*/$</searchRegex>
+              <replacement>/** Returns $1 */</replacement>
+            </replaceRegex>
             <!-- apply a specific flavor -->
```
Comment (virajjasani) [inline pom.xml:2739]: Is 2.23.0 bump required for specifying this regex or both (version bump + regex) are independent changes are we are combining for good?

**Review by apurtell — APPROVED** [review]
It's great there is an automated solution for this issue. Let's do it.

File: pom.xml (hunk 1 above)
Comment (apurtell) [reply pom.xml:2739]: Waiting for this question to be resolved before merging. I merged a PR that causes a conflict in FavoredNodesPlan. Happy to fix it up during the merge.

Comment (Apache9) [reply pom.xml:2739]: The version bumpiong is just a nice to have, not a must. If you think we'd better do it in a separated issue, we could open another one, not a big deal.

Comment (virajjasani) [reply pom.xml:2739]: Sounds good, I think combining with this PR was also fine, I just wanted to know whether this regex was going to be effective with version bump. But now that this PR is merged, let's bump it now with HBASE-27221. Thanks for this nice work @Apache9

**Pull request merged**